### PR TITLE
Mesh lc tests

### PR DESCRIPTION
**Pull request by mahaling**
Mesh based lens correction... 

working version with some validation to adapt to workflow. 

have added only basic integration tests. not complicated.

### CODE REVIEW

File: rendermodules/mesh_lens_correction/LensPointMatches.py [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+        ncpus = multiprocessing.cpu_count()
+    
+    args = {}
+    args['downsample_scale'] = downsample_scale
+    args['render'] = render 
+    args['match_collection'] = match_collection
+    args['nfeature_limit'] = nfeature_limit
+    args['matchMax'] = matchMax
+
+    pool = multiprocessing.Pool(ncpus)
```
Comment (RussTorres) [inline rendermodules/mesh_lens_correction/LensPointMatches.py]: Using `WithPool` defined in the renderapi client should make these calls testable without mocking non-forked processes.

File: rendermodules/mesh_lens_correction/LensPointMatches.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+    
+    fargs = []
+
+    if index_list is None:
+        index_list = range(tile_index.shape[0])
+    
+    for i in index_list:
+        impaths = [t.ip.mipMapLevels[0].imageUrl.split('file://')[-1]
```
Comment (RussTorres) [inline rendermodules/mesh_lens_correction/LensPointMatches.py]: Probably safer here to use the file uri handler in the form of `urllib.unquote(urlparse.urlparse(str(oldUrl)).path)`.

File: integration_tests/test_mesh_lens_correction.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    with open(js['output_json'], 'r') as f:
+        new_tform_dict = json.load(f)
+
+    with open(js['qc_json'], 'r') as f:
+        qc = json.load(f)
+
+    assert(not qc['hole_sections'] and not qc['gap_sections'])
+
+    #example_tform = renderapi.transform.ThinPlateSplineTransform(dataString=example_tform_dict['dataString'])
+    #new_tform = renderapi.transform.ThinPlateSplineTransform(dataString=new_tform_dict['dataString'])
```
Comment (RussTorres) [inline integration_tests/test_mesh_lens_correction.py]: How difficult is it to add a residual test comparison based on e.g. a grid?

File: rendermodules/mesh_lens_correction/LensPointMatches.py (hunk 1 above)
Comment (mahaling) [reply rendermodules/mesh_lens_correction/LensPointMatches.py]: Yes.. agreed but this and other changes will come in the next version of refactoring without changing the logic of the code... This PR is to just facilitate ease of processing for Tim.

File: integration_tests/test_mesh_lens_correction.py (hunk 3 above)
Comment (mahaling) [reply integration_tests/test_mesh_lens_correction.py]: yes planning to add this in next PR

Comment (RussTorres) [reply integration_tests/test_mesh_lens_correction.py]: I added issue #119 as a reminder so I feel okay about merging this.

**Pull request merged**